cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/generator/GenerationProcessorImpl.java

```diff
@@ -143,29 +149,36 @@ public GenerationReportTo generate(Object input, List<? extends GenerableArtifac
         this.targetRootPath = targetRootPath;
         generationReport = new GenerationReportTo();
 
+        progressCallback.accept("load Templates", 50);
         Collection<TemplateTo> templatesToBeGenerated = flatten(generableArtifacts);
 
         // generate
         Map<File, File> tmpToOrigFileTrace = Maps.newHashMap();
-        for (TemplateTo template : templatesToBeGenerated) {
-            try {
+        try {
+            for (TemplateTo template : templatesToBeGenerated) {
                 Trigger trigger = configurationHolder.readContextConfiguration().getTrigger(template.getTriggerId());
                 TriggerInterpreter triggerInterpreter = PluginRegistry.getTriggerInterpreter(trigger.getType());
                 InputValidator.validateTriggerInterpreter(triggerInterpreter, trigger);
                 tmpToOrigFileTrace.putAll(generate(template, triggerInterpreter));
-            } catch (CobiGenRuntimeException e) {
-                generationReport.setTemporaryWorkingDirectory(tmpTargetRootPath);
-                generationReport.addError(e);
-                LOG.error("An internal error occurred during generation.", e);
-            } catch (Throwable e) {
-                generationReport.setTemporaryWorkingDirectory(tmpTargetRootPath);
-                generationReport.addError(new CobiGenRuntimeException(
-                    "Something unexpected happened" + ((e.getMessage() != null) ? ": " + e.getMessage() : "!"), e));
-                LOG.error("An unknown exception occurred during generation.", e);
+                progressCallback.accept("generates... ", Math.round(1 / (float) templatesToBeGenerated.size() * 800));
             }
+        } catch (CobiGenCancellationException e) {
+            LOG.error("the Generation has been Canceled.", e);
+            generationReport.setCancelled(true);
+        } catch (CobiGenRuntimeException e) {
+            generationReport.setTemporaryWorkingDirectory(tmpTargetRootPath);
+            generationReport.addError(e);
+            LOG.error("An internal error occurred during generation.", e);
+        } catch (Throwable e) {
+            generationReport.setTemporaryWorkingDirectory(tmpTargetRootPath);
+            generationReport.addError(new CobiGenRuntimeException(
+                "Something unexpected happened" + ((e.getMessage() != null) ? ": " + e.getMessage() : "!"), e));
+            LOG.error("An unknown exception occurred during generation.", e);
         }
```

Comment:
Please keep the existing logic and abort anyhow by building another try catch around